### PR TITLE
doc: move main docs to book

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

The book has better formatting, search, discoverability. This is a start to move our documentation to one central place.

The matching PR for the book is here: https://github.com/tock/book/pull/23

However, documentation about code development and this repository makes sense to keep here. So documentation that is less about Tock as an operating system and more Tock as a software engineering exercise is kept here.

This is still a start so I think there is more to be done to figure out exactly where we want things.

### Testing Strategy

n/a

### TODO or Help Wanted

Some documents are still here as there is a different outstanding PR on this repo. Then I can move that doc.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +16,7 @@ device drivers as well as isolate device drivers from each other. Second, Tock
 uses memory protection units to isolate applications from each other and the
 kernel.
 
-[tock-ci]: https://github.com/tock/tock/actions?query=branch%3Amaster+workflow%3Atock-ci
+[tock-ci]: https://github.com/tock/tock/actions?query=branch%3Amaster+workflow%3Atock-ci\
```
Comment (bradjc) [inline README.md:19]: Need to revert this change.

**Comment by bradjc** [comment]
@brghena

**Review by brghena — APPROVED** [review]
